hack/ci/cache-wrapper.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# USAGE: cache-wrapper.sh hack/go_container.sh some-go-command
+# $@ will be executed with this script wrapping cache upload / download
+
+set -o errexit -o nounset -o pipefail
+
+# options for where the cache is stored
+BUCKET="${BUCKET:-bentheelder-kind-ci-builds}"
```

Comment:
I'm going to ask for a bucket from wg-k8s and move everything over, but this will do until that lands